### PR TITLE
- Model as having children, as opposed to being children.

### PR DESCRIPTION
**Pull request by baldersheim**
@geirst PR

### CODE REVIEW

File: searchlib/src/vespa/searchlib/query/streaming/query.cpp [hunk 1]
```diff
@@ -105,73 +120,76 @@ QueryConnector::create(ParseItem::ItemType type)
...
+    auto mt = getChildren().end();
+    if ((it != getChildren().end()) && (*it)->evaluate()) {
+        for (++it; it != mt; it++) {
+            if ((*it)->evaluate()) return false;
+        }
+        return true;
     }
-  }
-  return ok;
+    return false;
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/query/streaming/query.cpp:148]: I think this should be true, as _ok_ was set to true when empty before (line 125).

File: searchlib/src/vespa/searchlib/query/streaming/query.cpp [hunk 2]
```diff
@@ -105,73 +120,76 @@ QueryConnector::create(ParseItem::ItemType type)
...
-    //  Sanity check should be cheap enough that it does not matter.
-    for (const auto & child : *this) {
-        assert(dynamic_cast<const QueryTerm *>(child.get()) != nullptr);
-    }
     hl.clear();
     if ( !AndQueryNode::evaluate()) return hl;
 
     HitList tmpHL;
-    unsigned int numFields = size();
+    const QueryNodeList & children = getChildren();
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/query/streaming/query.cpp:185]: Consider using auto.

File: searchlib/src/vespa/searchlib/query/streaming/query.cpp [hunk 3]
```diff
@@ -219,13 +244,14 @@ PhraseQueryNode::evaluateHits(HitList & hl) const
     if ( ! AndQueryNode::evaluate()) return hl;
 
     HitList tmpHL;
-    unsigned int fullPhraseLen = size();
+    const QueryNodeList & children = getChildren();
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/query/streaming/query.cpp:247]: Consider using auto.

**Review by geirst — COMMENTED** [review]
👍 But one comment you must look at first.

File: searchlib/src/vespa/searchlib/query/streaming/query.cpp (hunk 3 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/query/streaming/query.cpp:247]: Fixed

File: searchlib/src/vespa/searchlib/query/streaming/query.cpp (hunk 2 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/query/streaming/query.cpp:185]: Fixed

File: searchlib/src/vespa/searchlib/query/streaming/query.cpp (hunk 1 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/query/streaming/query.cpp:149]: Fixed

**Review by geirst — COMMENTED** [review]
👍

**Pull request merged**